### DISCUSSION TITLE
Is it possible to inject hdr10+ generated metadata into the original file without extracting hevc file?

### DISCUSSION THREAD

**Discussion by wyup** [quietvoid/hdr10plus_tool#70] [Q&A] [answered]
Hello. I would like to inject .json hdr10+ metadata from a video file that somebody else generated with e.g. Resolve.
If I use hdr10plus_tool inject, I need to extract the hevc file, inject it and repack it. I wonder if I can inject this into the video file directly, with say ffmpeg. I have seen [this example](https://www.reddit.com/r/ffmpeg/comments/14bbelj/comment/joexim4/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) that extracts and injects hdr10+ metadata from and into a file using ffmpeg, but it appears to transcode video and I'd like to copy that, only muxing the .json metadata:

`ffmpeg -loglevel panic -i input.file -c:v copy -vbsf hevc_mp4toannexb -f hevc - | hdr10plus_tool extract -o output.json`

to extract from video file

`ffmpeg -fflags +discardcorrupt -ec guess_mvs+deblock+favor_inter -i input.file -c:a ac3 -ac 2 -ar 48000 -c:v libx265 -preset medium -x265-params 'hdr-opt=1:repeat-headers=1:colorprim=bt2020:transfer=smpte2084:colormatrix=bt2020nc:master-display=G(x,y)B(x,y)R(x,y)WP(x,y)L(x,y):max-cll=x,y:hdr10=1:dhdr10-info=output.json' -map_metadata -1 -map_chapters -1 output.file`

to inject it again into the file with colorimetry data from original file but transcoding with -preset medium. If I change to `-c:v copy` will it work? Because I won't be able to specify x265 colorimetry parameters. Are they needed by the way?

**Comment by quietvoid** [comment, marked as answer]
No, that specifically uses x265 to add back the metadata when reencoding.
You can't inject losslessly with FFmpeg, you need to extract the bitstream first.

**Reply by wyup** [reply]
Is it possible to pipe hdr10plus_tool inject to output from 'ffmpeg -i file.mkv -c copy -f hevc' to inject .json file into the bitstream, similar to extract command?

**Reply by quietvoid** [reply]
No, it's not supported.
